fdbserver/KeyValueStoreRocksDB.actor.cpp

```diff
@@ -1858,22 +1858,62 @@ struct RocksDBKeyValueStore : IKeyValueStore {
 	void set(KeyValueRef kv, const Arena*) override {
 		if (writeBatch == nullptr) {
 			writeBatch.reset(new rocksdb::WriteBatch());
+			setKeysSet.clear();
 		}
 		ASSERT(defaultFdbCF != nullptr);
 		writeBatch->Put(defaultFdbCF, toSlice(kv.key), toSlice(kv.value));
+		if (SERVER_KNOBS->ROCKSDB_SINGLEKEY_DELETES_ON_CLEARRANGE) {
+			setKeysSet.insert(kv.key);
+		}
+		TraceEvent("DBEventSet").detail("Key", kv.key.toString());
 	}
 
-	void clear(KeyRangeRef keyRange, const Arena*) override {
+	void clear(KeyRangeRef keyRange, const StorageServerMetrics* storageMetrics, const Arena*) override {
 		if (writeBatch == nullptr) {
 			writeBatch.reset(new rocksdb::WriteBatch());
+			setKeysSet.clear();
```

Comment:
This a bug fix?

Reply:
We want a new setKeys whenever writeBatch is empty to start fresh